consensus/cuckoo/consensus.go

```diff
@@ -301,7 +301,7 @@ func (cuckoo *Cuckoo) verifyHeader(chain consensus.ChainReader, header, parent *
 		return consensus.ErrInvalidNumber
 	}
 
-	if header.Quota.Cmp(new(big.Int).Add(parent.Quota, new(big.Int).SetUint64(chain.Config().GetBlockQuota(header.Number)))) != 0 {
+	if header.Quota < parent.Quota || header.Quota != parent.Quota+chain.Config().GetBlockQuota(header.Number) {
```

Comment:
Whether the second condition contains first one?

Reply:
The first one consider the overflow of uint64.
For example, parent.Quota = MaxUint64 - 1,  blockQuota = 2, header.Quota = 1. This situation satisfies the second condition but dissatisfies the first one.

Reply:
Can quota reach MaxUint64, we should avoid overflow in quota calculation.
In fact we can calculate when the quota limit will be

Reply:
Yeah, gasLimit is set for each block own, but the quota is accumulated through the whole blockchain.
For the max quota value in each block is 65536 * 128 (for dolores), which means the limit will be reached at block number 2199023255551

Reply:
Yes, gas limit from big.int to uint64 because they know the ceiling https://github.com/ethereum/go-ethereum/commit/6f69cdd109b1dd692b8dfb15e7c53d2051fbc946#diff-afbbe5ab9727e568254cfee647e62876

2199023255551 is huge for block number, should we consider it as ∞

Reply:
If we can consider it as ∞, I think we can also convert the quota into uint64